kansha/card_addons/gallery/comp.py

```diff
@@ -44,26 +45,63 @@ def __init__(self, card, action_log, assets_manager_service):
         """
         super(Gallery, self).__init__(card, action_log)
         self.assets_manager = assets_manager_service
-        self.data = DataGallery(self.card)
-
         self.assets = []
-        for data_asset in self.data.get_assets():
-            self._create_asset_c(data_asset)
+        self.load_assets()
         self.comp_id = str(random.randint(10000, 100000))
         self.model = 'view'
         self.cropper = component.Component()
 
+    def load_assets(self):
+        for asset_data in DataAsset.get_all(self.card.data):
+            self.create_asset(asset_data)
+
+    def delete_asset(self, asset):
+        """Delete asset
+
+        Delete asset from card (component and overlay), database
+        and asset manager.
+
+        In:
+            - ``asset`` -- Asset to delete
+        """
+        for a in self.assets:
+            if a() is asset:
+                self.assets.remove(a)
+                break
+        DataAsset.remove(self.card.data, asset.filename)
+        asset.delete()
+
+    def delete(self):
+        for asset in self.assets:
+            self.assets_manager.delete(asset().filename)
+        DataAsset.remove_all(self.card.data)
+        self.assets = []
+
     def copy(self, parent, additional_data):
         new_extension = self.__class__(parent, parent.action_log, self.assets_manager)
-        for data_asset in self.data.get_assets():
+        for data_asset in DataAsset.get_all(self.card.data):
             new_asset_id = self.assets_manager.copy(data_asset.filename)
```

Comment:
Use the assets manager of Kansha v2 where the assets are active objects with an hidden reference to the assets manager